### PR TITLE
Add two-sample Kolmogorov-Smirnov test

### LINKED ISSUE #1138: Add two-sample Kolmogorov-Smirnov test
### Goal
Add ran.test.kolmogorovSmirnov(x, y) to compute the two-sample Kolmogorov-Smirnov test statistic D = sup|F1(x) - F2(x)| over the pooled empirical CDFs of two samples, along with an asymptotic p-value using the existing Kolmogorov distribution.

### Scope
- Add `src/test/kolmogorov-smirnov.js` with the two-sample KS test implementation
- Export kolmogorovSmirnov from `src/test/index.js`
- Add test cases to `test/test-ks-two-sample.js`

### Acceptance Criteria
- [ ] Implement kolmogorovSmirnov(x, y, opts) computing D statistic and asymptotic p-value via Kolmogorov.cdf()/survival()
- [ ] Handle edge cases: empty arrays, single-element samples, identical samples
- [ ] Add comprehensive test cases verifying D statistic calculation and p-value against reference implementations (scipy.stats.ks_2samp or R ks.test)
- [ ] Integrate test into the test suite and confirm all tests pass
- [ ] Production-code diff under 400 lines (excluding tests). If exceeded, decompose.
- [ ] PR is independently revertable without breaking main
- [ ] Prerequisites (new special functions, algorithms) filed as separate issues and merged first

### Out of Scope
- One-sample KS test (already exists as ran.test.kolmogorov)
- Exact p-value computation via permutation methods
- Bootstrap-based p-values

### PR DESCRIPTION
**Pull request by synesenom**
## Summary

Adds `ran.test.kolmogorovSmirnov(x, y, alpha)`, the two-sample Kolmogorov-Smirnov test. The statistic `D = sup|F1(x) - F2(x)|` is computed over the pooled empirical CDFs of the two samples (binary search into each sorted sample at every pooled point); the asymptotic p-value comes from the existing `ran.dist.Kolmogorov` distribution's `survival()`, evaluated at `sqrt(n1*n2/(n1+n2))·D`.

Closes #1138

## Design decisions

- [ADR-0042: Return shape for single-sample goodness-of-fit tests in `ran.test`](decisions/0042-single-sample-gof-test-return-shape.md) — explicitly names this issue (#1138) as required to follow `{stat, passed, pValue}`, so this PR returns that shape rather than the plain `{stat, passed}` shape used by the module's older multi-sample tests (`mannWhitney`, `welch`).

## Non-trivial changes

- **`src/test/kolmogorov-smirnov.js`** (new file): the D-statistic computation and asymptotic p-value formula. Worth a close look at the reference values in the new precision tests — the p-value is deliberately cross-checked against `scipy.stats.kstwobign.sf(sqrt(ne)*D)` (the pure n→∞ asymptotic Kolmogorov distribution, matching `ran.dist.Kolmogorov` and the issue's explicit spec), **not** against `scipy.stats.ks_2samp(...).pvalue`'s default — scipy's default instead reports a more precise finite-sample-corrected distribution (`kstwo` via Marsaglia-Wong-Wei) that isn't implemented in ranjs and is out of scope for this issue.

## Testing

- `test/test.js` — new `describe('kolmogorovSmirnov', ...)` block: empty-array throws, identical samples (stat=0, pValue=1), three precision cases cross-checked against scipy 1.17.1 covering equal sample sizes, unequal sample sizes, and tied values, plus pass/reject cases against matching/mismatched discrete (Poisson) and continuous (Normal) distributions.
- `CHANGELOG.md` — `### Added` bullet under `## [Unreleased]`.

## Checklist

- [x] `npm run standard` passes (no linter errors)
- [x] `npm test` passes (all tests green — 8222 passing, coverage thresholds met)
- [x] `npm run typecheck` passes
- [x] Tests added for new behavior
- [x] `CHANGELOG.md` updated under `## [Unreleased]`
- [x] Production-code diff is under ~400 lines (~82 lines: 81-line new file + 1 export line)
- [x] `/review`'s 8-agent pass ran clean (no Blocks); 3 test-coverage Warns fixed inline, 1 doc Warn filed as #1174 (pre-existing, unrelated `cramerVonMises` field-order nit), 2 style/perf Warns triaged as not-bugs

---
*Generated with [Claude Code](https://claude.ai/code)*

---
_Generated by [Claude Code](https://claude.ai/code/session_01LPcjDbH4hy17w3W4hLiu78)_

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -198,6 +198,102 @@ describe('test', () => {
...
+    })
+
+    it('should pass for samples of the same continuous distribution', () => {
+      seed(0)
+      const mu = float(0, 5)
+      const sigma = float(1, 10)
+      const sample1 = (new Normal(mu, sigma)).seed(1).sample(SAMPLE_SIZE)
+      const sample2 = (new Normal(mu, sigma)).seed(2).sample(SAMPLE_SIZE)
+      assert(test.kolmogorovSmirnov(sample1, sample2).passed)
+    })
```
Comment (synesenom) [inline test/test.js]: Confirmed `main`'s `test.js` scores a clean 10.0, so this duplication was genuinely introduced by this PR's two new tests matching the existing `mannWhitney`/`welch` "pass for same distribution / reject for different distribution" idiom closely enough to tip the detector's threshold. Fixed in 132d085 by merging the two near-duplicate `kolmogorovSmirnov` tests into one (same behavioral coverage — pass for identical-distribution samples, reject for shifted samples — just no longer structurally duplicating the sibling blocks). `test.js` is back to a 10.0 Code Health score.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by synesenom** [comment]
CI red on `test (22)` — but the only failure is `cramerVonMises > should pass for samples drawn from the tested distribution`, a pre-existing flaky test tracked separately in #1172 (global PRNG state race under `mocha --parallel`, unrelated to this PR — every `kolmogorovSmirnov` test in this run passed). Re-running the job now; will act on the recovery notice if it lands, or re-run again if it flakes again.

---
_Generated by [Claude Code](https://claude.ai/code)_
